### PR TITLE
Basic <script> and data URL test

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/html/pull/1782 for pointers to more
context.

### CODE REVIEW

**Comment by annevk** [comment]
I can't quite get the `setup()` thing to work. Anyone any idea what I did wrong?

**Comment by annevk** [comment]
Should be all good now.

File: html/semantics/scripting-1/the-script-element/data-url.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>data URL and scripts</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+<script>
+  setup({allow_uncaught_exception:true})
+  var t = async_test()
+  t.step(() => {
```
Comment (foolip) [inline html/semantics/scripting-1/the-script-element/data-url.html]: `async_test(t => { ... })` saves a line and one need not look to see if `t` is referenced elsewhere.

File: html/semantics/scripting-1/the-script-element/data-url.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    var counter = 1
+    window.onerror = t.step_func((message, x, xx, xxx, e) => {
+      assert_equals(typeof e, "number")
+      assert_equals(e, counter)
+      assert_not_equals(message, "Script error.") // Cannot be "muted" as data URLs are same-origin
+      if(counter == 3) {
```
Comment (foolip) [inline html/semantics/scripting-1/the-script-element/data-url.html]: More whitespace

**Review by foolip — CHANGES_REQUESTED** [review]

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Hmm, https://w3c-test.org/submissions/3945/html/semantics/scripting-1/the-script-element/data-url.html actually fails on all but Firefox.

**Comment by annevk** [comment]
Yeah, only Fx treats it as same-origin like it should.

**Comment by foolip** [comment]
But the failures are because of other differences first. Edge only gives 4 arguments on the `onerror` callback, while Chrome and Safari make the 5th argument an object instead of number.

The spec is at https://html.spec.whatwg.org/#the-event-handler-processing-algorithm and Edge is wrong about the number of arguments, not sure about the type though, that probably affects `window.addEventListener('error', ...)` too.

**Comment by annevk** [comment]
Pretty sure that's because Chrome and Safari treat these as cross-origin and therefore create the Error object with "Script error." as a message.

**Comment by youennf** [comment]
I would think so as well.
WebKit might change that behavior soon so these tests are most welcome

**Comment by foolip** [comment]
Ah, yes, the "Script error." check fails if those before it are commented out. Will change the order to make it obvious that it's a real failure to someone investigating the failure.

**Comment by foolip** [comment]
@mikewest if you want to check this after the fact that'd be nice, but I've looked at it enough to be comfortable merging now.

**Pull request merged**